compiler/src/dotty/tools/dotc/core/Scopes.scala

```diff
@@ -204,6 +210,12 @@ object Scopes {
      */
     private var elemsCache: List[Symbol] = null
 
+    /** The synthesizer to be used, or `null` if no synthesis is done on this scope */
+    private var synthesize: SymbolSynthesizer = null
```

Comment:
Note that with both scalac and doty, not using `private[this]` here will generate a pair of getter and setter, this is arguably something that we should fix in the compiler since `private[this]` and `private` should be considered equivalent in an object.